src/lib/extra-firmware.ts

```diff
@@ -0,0 +1,77 @@
+import { TypedError } from 'typed-error';
+
+import type ConfigJsonConfigBackend from '../config/configJson';
+import { InternalInconsistencyError, isNotFoundError } from './errors';
+import log from './supervisor-console';
+import { docker } from './docker-utils';
+import constants from './constants';
+
+export const EXTRA_FIRMWARE_VOLUME_NAME = 'extra-firmware';
+
+export class ExtraFirmwareVolumeError extends TypedError {
+	constructor() {
+		super(`Found extra firmware volume: ${EXTRA_FIRMWARE_VOLUME_NAME}`);
+	}
+}
+
+export const initialize = async (configJson: ConfigJsonConfigBackend) => {
+	log.info('Configuring extra firmware volume');
+	// Create volume - volume will be empty unless user includes
+	// appropriate linux-firmware block in their docker-compose.
+	// The OS is configured to use the volume path on host for firmware,
+	// but it's on the user to include the firmware block as well as reboot.
+	await create();
+
+	// Configure OS to look for extra firmware volume
+	await configureVolume(configJson);
+};
+
+const configureVolume = async (configJson: ConfigJsonConfigBackend) => {
+	const os: { [key: string]: any } = (await configJson.get('os')) ?? {};
+	if (typeof os !== 'object') {
+		throw new InternalInconsistencyError(
```

Comment:
I don't think this qualifies as an `InternalConsistencyError`, that should be only to validate that internal calls are correct.

I also don't think a misconfigured config.json should crash the supervisor here, we probably need a warning or maybe issue a system error (that shows up in the dashboard)